OneCommunityGlobal/HighestGoodNetworkApp
# Tim: Tool Details View Remake

# Description
Remake of Tool Details view pull request created by Olga @cvtqx that was merged and then reverted. All coding in this PR is Olga's work, with the exception of a minor bug fix and name changes.

## Related PR:
[Backend PR 698](https://github.com/OneCommunityGlobal/HGNRest/pull/698)

## Main changes explained:
- created components: ToolModal.jsx, ToolDetailPage.jsx
- added ToolDetailPage.css for styling
- added tool actions, reducer and route

## How to test:

1. [Follow instructions to start backend](https://github.com/OneCommunityGlobal/HGNRest/pull/698)
2. `git checkout Tim_tool_details_view_remake`
3. `npm i`, then `npm run start:local`
4. Log in as volunteer user
5. Navigate to http://localhost:3000/bmdashboard/login, verify credentials
6. Navigate to http://localhost:3000/bmdashboard/tools/659f72a1e49bf93bc79a1b2d to view a tool with 'Purchase' status
7. Navigate to http://localhost:3000/bmdashboard/tools/659f72c0e49bf93bc79a1b2e to view a tool with 'Rental' status
8. Check that the links are working. Link to Buy/Rent takes you to a new page (HomeDepot website at the moment). Description link opens a modal with tool description.
9. Check that if the tools Ownership is Purchase, Rental Duration line is not displayed.
10. Check the page in different screen sizes.

## Note:
Please remember to leave reviews on both front and backend PRs!
